pkg/controller/computenodeopenstack/computenodeopenstack_controller.go

```diff
@@ -395,45 +483,114 @@ func updateNodesStatus(c client.Client, instance *computenodev1alpha1.ComputeNod
 				err = deleteBlockerPodFinalizer(c, node.Name, instance)
 			}
 		case "Ready":
+			log.Info("Previous Status Ready: " + node.Name)
 			newNode := computenodev1alpha1.Node{Name: node.Name, Status: nodeStatus}
 			nodesStatus = append(nodesStatus, newNode)
 
 			if nodeStatus == "Ready" {
 				readyWorkers += 1
 			} else if nodeStatus == "SchedulingDisabled" {
-				// trigger drain (new pod)
-				err = triggerNodeDrain(c, node.Name, instance)
+				// trigger node drain pre tasks (new pod)
+				err = triggerNodeDrain(c, node.Name, instance, true, openstackClientAdmin, openstackClient)
 				if err != nil {
 					return err
 				}
 			}
 		case "SchedulingDisabled":
-			// if drain has finished (pod completed) remove blocker-pod finalizer
-			drained, err := isNodeDrained(c, node.Name, instance)
-			if err != nil {
+			log.Info(fmt.Sprintf("Previous Status SchedulingDisabled: %s", node.Name))
+			// if pre draining has finished (job completed) remove blocker-pod finalizer
+			drainedPre, err := isNodeDrained(c, node.Name, node.Name+"-drain-job-pre", instance)
+			if err != nil && errors.IsNotFound(err) {
+				log.Error(err, fmt.Sprintf("Pre NodeDrain job errors.IsNotFound: %s", node.Name+"-drain-job-pre"))
+				ResyncPeriod = 30 * time.Second
+				return nil
+			} else if err != nil {
 				return err
 			}
-			if drained {
-				err = deleteBlockerPodFinalizer(c, node.Name, instance)
+
+			log.Info(fmt.Sprintf("Pre draining job: %s", drainedPre))
+			switch drainedPre {
+			case "succeeded":
+				// taint node that daemonset pods get stopped
+				nodes, err := kclient.CoreV1().Nodes().List(metav1.ListOptions{})
+
+				for _, updateNode := range nodes.Items {
```

Comment:
I don't get this, you already have the node name and the node object. Is this to ensure the information is updated from before the loop to inside the loop? In addition, we should not list all the nodes, only the ones associated to our worker-role